nil.go

```diff
@@ -0,0 +1,64 @@
+package pack
+
+import (
+	"math/rand"
+	"reflect"
+)
+
+// Nil represents a nil value.
+type Nil struct{}
+
+// NewNil wraps an existing raw struct.
+func NewNil(x struct{}) Nil {
+	return Nil(x)
+}
```

Comment:
Why does it wrap an existing empty struct? Why not:
```suggestion
// NewNil wraps an existing raw struct.
func NewNil() Nil {
	return Nil{}
}
```